cobalt/user_agent_platform_info.h

```diff
@@ -0,0 +1,140 @@
+// Copyright 2024 The Cobalt Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef COBALT_USER_AGENT_PLATFORM_INFO_H_
+#define COBALT_USER_AGENT_PLATFORM_INFO_H_
+
+#include <map>
+#include <string>
+
+namespace cobalt {
+
+void GetUserAgentInputMap(
+    const std::string& user_agent_input,
+    std::map<std::string, std::string>& user_agent_input_map);
+
+class UserAgentPlatformInfo {
+ public:
+  explicit UserAgentPlatformInfo(bool enable_skia_rasterizer = false);
+  ~UserAgentPlatformInfo() = default;
+
+  const std::string ToString() const;
```

Comment:
Returning a `const std::string` makes me think that the returned thing is immutable (probably pointing underneath to some UserAgentPlatformInfo data). If this is not the case, consider returning a non-const std::string.